src/Lykke.Job.BlockchainOperationsExecutor/Workflow/CommandHandlers/ForgetBroadcastedTransactionCommandsHandler.cs

```diff
@@ -13,20 +14,36 @@ public class ForgetBroadcastedTransactionCommandsHandler
     {
         private readonly IChaosKitty _chaosKitty;
         private readonly IBlockchainApiClientProvider _apiClientProvider;
+        private readonly ICapabilitiesService _capabilitiesService;
+        private readonly ISourceAddresLocksRepoistory _locksRepoistory;
 
         public ForgetBroadcastedTransactionCommandsHandler(
             IChaosKitty chaosKitty,
-            IBlockchainApiClientProvider apiClientProvider)
+            IBlockchainApiClientProvider apiClientProvider,
+            ICapabilitiesService capabilitiesService,
+            ISourceAddresLocksRepoistory locksRepoistory)
         {
             _chaosKitty = chaosKitty;
             _apiClientProvider = apiClientProvider;
+            _capabilitiesService = capabilitiesService;
+            _locksRepoistory = locksRepoistory;
         }
 
         [UsedImplicitly]
         public async Task<CommandHandlingResult> Handle(ForgetBroadcastedTransactionCommand command, IEventPublisher publisher)
         {
             var apiClient = _apiClientProvider.Get(command.BlockchainType);
 
+            var capabilities = await _capabilitiesService.GetAsync(command.BlockchainType);
+            if (capabilities.IsExclusiveWithdrawalsRequired)
+            {
+                await _locksRepoistory.ReleaseLockAsync(command.BlockchainType, command.FromAddress, command.OperationId);
+                _chaosKitty.Meow(command.OperationId);
+
+                await _locksRepoistory.ReleaseLockAsync(command.BlockchainType, command.ToAddress, command.OperationId);
```

Comment:
When nested locks are used, order of the releasing should be opposit to the order of the locking. Looks like this doesn't matter here, but in multithreading better to respect rules.